web/themes/contrib/civictheme/civictheme_starter_kit/build.js

```diff
@@ -0,0 +1,266 @@
+// phpcs:ignoreFile
+/*
+Civictheme build - version 0.0.1 (alpha).
+
+This is a simplified build system for civictheme, designed to:
+
+- Speed up build time
+- Improve auditability by reducing dependencies / external code
+
+Build by calling:
+
+  npm run build:new
+
+Build and watch by calling:
+
+  npm run build:watch:new
+
+Notes:
+
+- Requires Node version 22 (for the built in glob feature).
+- No Windows support (unless through WSL).
+- This relies on command line rsync.
+- This does not support uglify or minifying JS / CSS code.
+- Watch does not trigger on a directory change, only on a (scss, twig, js) file change.
+*/
+
+const fs = require('fs')
+const path = require('path')
+const { globSync } = require('node:fs')
+const { execSync } = require('child_process')
+const sass = require('sass')
+
+// ----------------------------------------------------------------------------- UTILITIES
+
+function runCommand(command) {
+  execSync(command, (error, stdout, stderr) => {
+    if (error) {
+      console.log(`error: ${error.message}`)
+      return
+    }
+    if (stderr) {
+      console.log(`stderr: ${stderr}`)
+      return
+    }
+    if (stdout) {
+      console.log(stdout)
+    }
+  })
+}
+
+function getStyleImport(path, cwd) {
+  return globSync(path, { cwd }).sort((a, b) => a.localeCompare(b)).map(i => `@import "${i}";`).join('\n')
+}
+
+function loadStyle(path, cwd) {
+  const result = []
+  let data = fs.readFileSync(path, 'utf-8')
+
+  data.split('\n').forEach(line => {
+    // Only glob imports with *!()| characters.
+    const match = /@import '(.*[\*!()|].*)'/.exec(line)
+    result.push(match ? getStyleImport(match[1], cwd) : line)
+  })
+
+  return result.join('\n')
+}
+
+function fullPath(filepath) {
+  return path.resolve(PATH, filepath)
+}
+
+// ----------------------------------------------------------------------------- CONFIGURATION
+const config = {
+  build: false,
+  watch: false,
+  combine: false,
+  styles: false,
+  styles_editor: false,
+  styles_admin: false,
+  styles_layout: false,
+  styles_variables: false,
+  js: false,
+  assets: false,
+  base: false,
+}
+
+const flags = process.argv.slice(2)
+const buildType = ['build', 'watch']
+const buildWatchFlagCount = flags.filter(f => buildType.indexOf(f) >= 0).length
+
+if (flags.length <= 2 && buildWatchFlagCount <= 2) {
+  // If build and/or watch, or neither..
+  config.build = buildWatchFlagCount === 0 || flags.indexOf('build') >= 0
+  config.watch = flags.indexOf('watch') >= 0
+  config.combine = true
+  config.styles = true
+  config.styles_editor = true
+  config.styles_variables = true
+  config.js = true
+  config.assets = true
+} else {
+  // Fully configured from command line - everything disabled by default.
+  flags.forEach((flag) => {
+    config[flag] = true
+  })
+}
+
+// ----------------------------------------------------------------------------- PATHS
+
+const PATH = __dirname
+
+const THEME_NAME              = PATH.split('/').reverse()[0]
```

Comment:
i think adding comments for each variable will really help everyone to understand the purpose